risc0/zkvm/src/prove/mod.rs

```diff
@@ -213,6 +217,12 @@ pub trait Prover {
 }
 
 fn provers() -> HashMap<String, Rc<dyn Prover>> {
+    if cfg!(feature = "disable-dev-mode") && std::env::var("RISC0_DEV_MODE").is_ok() {
```

Comment:
@SchmErik sorry I'm a bit late to the party, but I am reading through this now and have a concern.

`std::env::var("RISC0_DEV_MODE").is_ok()` has some unfortunate semantics such as `RISC0_DEV_MODE=false` and `RISC0_DEV_MODE=''` both yielding the system in dev mode. My impression is most developers have never actually `unset` and environment variable because in most cases, setting it to empty string is almost always treated as semantically equivalent. Can we update this to avoid these sharp edges? One appraoch that would seem pretty clear would be to have `true` and `false` (and possibly others like `1` and `0`) turn dev mode on and off respectively, unset would generally be "off", and other values would be an error (e.g. `RISC0_DEV_MODE=sure` bails).
 
 One other reason for this suggestion is that in test environments it may actually make sense to default to dev mode, and set `RISC0_DEV_MODE=false` to get proving to occur. This suggestion is less important and I am not 100% convinced this is the right way to do it.

Reply:
(Oh, and another weirdness about this line is that if the `Err` is something other than `VarError::NotPresent` then this will treat it as unset)

Reply:
@nategraf good catch! I'll address this in a follow-up PR